### ISSUE TITLE
new warnings during charon configuration

### ISSUE THREAD

**Issue by glhenni** [TriBITSPub/TriBITS#382]
We're seeing the included warning, previously unseen, during our configuration/cmake step for charon. It's only a warning and we could add the flag it mentions to suppress it but you might want to take a look at it.
```
CMake Warning (dev) in /scratch/charonops/jenkins/workspace/ASCIC_Build_Farm_CDE-GNU-dbg/TriBITS/tribits/common_tpls/utils/ParseLibraryList.cmake:
  A logical block opening on the line

    /scratch/charonops/jenkins/workspace/ASCIC_Build_Farm_CDE-GNU-dbg/TriBITS/tribits/common_tpls/utils/ParseLibraryList.cmake:51 (function)

  closes on the line

    /scratch/charonops/jenkins/workspace/ASCIC_Build_Farm_CDE-GNU-dbg/TriBITS/tribits/common_tpls/utils/ParseLibraryList.cmake:115 (endfunction)

  with mis-matching arguments.
Call Stack (most recent call first):
  /scratch/charonops/jenkins/workspace/ASCIC_Build_Farm_CDE-GNU-dbg/TriBITS/tribits/common_tpls/utils/AddPackageDependency.cmake:51 (include)
  /scratch/charonops/jenkins/workspace/ASCIC_Build_Farm_CDE-GNU-dbg/TriBITS/tribits/common_tpls/find_modules/FindNetCDF.cmake:72 (include)
  /scratch/charonops/jenkins/workspace/ASCIC_Build_Farm_CDE-GNU-dbg/TriBITS/tribits/common_tpls/FindTPLNetcdf.cmake:66 (find_package)
  /scratch/charonops/jenkins/workspace/ASCIC_Build_Farm_CDE-GNU-dbg/TriBITS/tribits/core/package_arch/TribitsProcessEnabledTpl.cmake:106 (include)
  /scratch/charonops/jenkins/workspace/ASCIC_Build_Farm_CDE-GNU-dbg/TriBITS/tribits/core/package_arch/TribitsGlobalMacros.cmake:1604 (tribits_process_enabled_tpl)
  /scratch/charonops/jenkins/workspace/ASCIC_Build_Farm_CDE-GNU-dbg/TriBITS/tribits/core/package_arch/TribitsProjectImpl.cmake:196 (tribits_process_enabled_tpls)
  /scratch/charonops/jenkins/workspace/ASCIC_Build_Farm_CDE-GNU-dbg/TriBITS/tribits/core/package_arch/TribitsProject.cmake:93 (tribits_project_impl)
  CMakeLists.txt:61 (TRIBITS_PROJECT)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) in /scratch/charonops/jenkins/workspace/ASCIC_Build_Farm_CDE-GNU-dbg/TriBITS/tribits/common_tpls/utils/AddPackageDependency.cmake:
  A logical block opening on the line

    /scratch/charonops/jenkins/workspace/ASCIC_Build_Farm_CDE-GNU-dbg/TriBITS/tribits/common_tpls/utils/AddPackageDependency.cmake:53 (function)

  closes on the line

    /scratch/charonops/jenkins/workspace/ASCIC_Build_Farm_CDE-GNU-dbg/TriBITS/tribits/common_tpls/utils/AddPackageDependency.cmake:175 (endfunction)

  with mis-matching arguments.
Call Stack (most recent call first):
  /scratch/charonops/jenkins/workspace/ASCIC_Build_Farm_CDE-GNU-dbg/TriBITS/tribits/common_tpls/find_modules/FindNetCDF.cmake:72 (include)
  /scratch/charonops/jenkins/workspace/ASCIC_Build_Farm_CDE-GNU-dbg/TriBITS/tribits/common_tpls/FindTPLNetcdf.cmake:66 (find_package)
  /scratch/charonops/jenkins/workspace/ASCIC_Build_Farm_CDE-GNU-dbg/TriBITS/tribits/core/package_arch/TribitsProcessEnabledTpl.cmake:106 (include)
  /scratch/charonops/jenkins/workspace/ASCIC_Build_Farm_CDE-GNU-dbg/TriBITS/tribits/core/package_arch/TribitsGlobalMacros.cmake:1604 (tribits_process_enabled_tpl)
  /scratch/charonops/jenkins/workspace/ASCIC_Build_Farm_CDE-GNU-dbg/TriBITS/tribits/core/package_arch/TribitsProjectImpl.cmake:196 (tribits_process_enabled_tpls)
  /scratch/charonops/jenkins/workspace/ASCIC_Build_Farm_CDE-GNU-dbg/TriBITS/tribits/core/package_arch/TribitsProject.cmake:93 (tribits_project_impl)
  CMakeLists.txt:61 (TRIBITS_PROJECT)
This warning is for project developers.  Use -Wno-dev to suppress it.
```

**Comment by bartlettroscoe** [comment]
Looks like there are several of these:

```
[rabartl@crf450 TriBITS (master)]$ find tribits/ -name "*.cmake" -exec grep -nHF "endfunction(" {} \; | grep -vF "endfunction()"
tribits/common_tpls/find_modules/FindHDF5.cmake:90:endfunction(_HDF5_CHOMP_STRING)
tribits/common_tpls/find_modules/FindHDF5.cmake:118:endfunction(_HDF5_PARSE_SETTINGS_FILE)
tribits/common_tpls/find_modules/FindHDF5.cmake:129:endfunction(_HDF5_DEFINE_VERSION _var)
tribits/common_tpls/find_modules/FindHDF5.cmake:141:endfunction(_HDF5_DEFINE_PARALLEL_BUILD _file _var)
tribits/common_tpls/find_modules/FindHDF5.cmake:174:endfunction(_HDF5_EXTRA_LIBRARY_DIRS _file _var)
tribits/common_tpls/find_modules/FindHDF5.cmake:184:endfunction(_HDF5_LIBRARY_PATH _file _var)
tribits/common_tpls/find_modules/FindHDF5.cmake:234:endfunction(_HDF5_EXTRA_LIBRARIES _file _var)
tribits/common_tpls/find_modules/FindHDF5.cmake:278:endfunction(_HDF5_EXTRA_INCLUDE_DIRS _file _var)
tribits/common_tpls/utils/AddPackageDependency.cmake:175:endfunction(ADD_PACKAGE_DEPENDENCY)
tribits/common_tpls/utils/AddImportedLibrary.cmake:114:endfunction(ADD_IMPORTED_LIBRARY)
tribits/common_tpls/utils/ParseLibraryList.cmake:115:endfunction(PARSE_LIBRARY_LIST)
```

That missed my refactoring to lower-case.  (Do not realize anyone actually used this.  This is universally considered undesirable; see [here](https://community.kde.org/Policies/CMake_Coding_Style#End_commands) and [here](http://docs.ros.org/en/jade/api/catkin/html/user_guide/standards.html)).

I will push a commit that fixes this.

**Comment by bartlettroscoe** [comment]
@glhenni, FYI, PR #383 should have this fixed.  I am just waiting for @gsjaardema to review and approve that fix.

**Comment by bartlettroscoe** [comment]
@glhenni, I went ahead and merged PR #383.  @gsjaardema can do a post-merge review of #383 if he likes.  Otherwise, I suspect this will solve the problem.  If this does not solve the problem, please re-open this issue.
